src/test/java/com/google/sps/CommentRetrievalTest.java

```diff
@@ -0,0 +1,149 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.mock;
+
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.CommentThread;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.sps.servlets.utils.YouTubeCommentRetriever;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+// Using wildcard import since documentation is unclear which package
+// "USES_DEEP_STUBS" is imported from
+
+@RunWith(JUnit4.class)
+public class CommentRetrievalTest {
+
+  private final int HUNDRED = 100;
+  private final int EXCESS_HUNDRED = 200;
+  private final int MAX_COMMENTS_PER_TOKEN = 100;
+  private YouTubeCommentRetriever commentRetriever;
+  // These url's no longer do anything; rather, serve as parameter for the call
+  private final String POPULAR_VIDEO_URL = "dQw4w9WgXcQ";
+  private final String UNPOPULAR_VIDEO_URL = "cA-arJ0T6L4";
+  private final String NEXT_PAGE_TOKEN = "Some Page Token";
+  private final String FIRST_TOKEN_COMMENT = "First";
+  private final String SECOND_TOKEN_COMMENT = "Second";
+
+  public void setUp(int numExpectedComments) throws Exception {
+    setUp(numExpectedComments, null);
+  }
+  // Simulate a CommentThreadListResponse with exactly as many comments as expected
+  public void setUp(int numExpectedComments, String nextPageToken) throws Exception {
```

Comment:
This method can be named more expressively since `setUp` is a little vague.

Maybe something like `setupYoutubeMocksToReturnNumComments`? Open to anything here.